NewHaven/src/Setting.cpp

```diff
@@ -0,0 +1,188 @@
+    board = nullptr;
+    players = nullptr;
+    current = this;
+}
+
+Setting::Setting(const Setting& setting):h_deck{new HarvestDeck(*setting.h_deck)}, b_deck{new BuildingDeck(*setting.b_deck)},
+    board{new GBMap(*setting.board)}, players{new vector<Player>(*setting.players)}{
+    // singleton design we dont need any other reference but the current one
+    delete current;
+    current = this;
+}
+
+Setting& Setting::operator=(const Setting& setting){
+    if(this == &setting)
+        return *this;
+
+    // cannot assume that the value ares nullptr and cant use assignment operator for Harvest/Building deck since implicitly deleted
+   delete h_deck;
+   delete b_deck;
+   delete board;
+   delete players;
+    h_deck = new HarvestDeck(*setting.h_deck);
+    b_deck = new BuildingDeck(*setting.b_deck);
+    board = new GBMap(*setting.board);
+    players = new vector<Player>(*setting.players);
+    current = this;
+    return *this;
+};
+Setting::~Setting() {
+    delete h_deck;
+    delete b_deck;
+    delete board;
+    delete players;
+    current = nullptr;
+}
+
+void Setting::setupPlayers(const int numberOfPlayers) {
+    std::cout << "CREATING " << numberOfPlayers << " PLAYERS!" << endl;
+    if(players == nullptr)
+        players = new vector<Player>;
+    players->clear();
+    for(int i = 0; i < numberOfPlayers;i++){
+        players->push_back(Player());
+    }
+    std::cout << numberOfPlayers << " PLAYERS HAVE BEEN SUCCESSFULLY CREATED!" << endl;
+}
+
+void Setting::loadGameBoard(const std::string filepath) {
+    cout << "LOADING " << filepath << endl;
+    GBMapLoader loader;
+    if(loader.loadConfig(filepath) && board == nullptr) {
+        cout << "LOADING SUCCESSFUL" << endl;
+        board = loader.generateMap();
+    }
+    else{
+    cout << "LOADING FAILED! AN ERROR HAS OCCURRED" << endl;
+    throw BoardConfigurationNotLoaded();
+    }
+}
+
+VGMap* Setting::loadVillageMap(const std::string filepath) {
+    cout << "LOADING " << filepath << endl;
+    VGMapLoader loader;
+    if(loader.loadVConfig(filepath)) {
+        cout << "LOADING SUCCESSFUL" << endl;
+        return loader.generateVMap();
+    }
+    cout << "LOADING FAILED! AN ERROR HAS OCCURRED" << endl;
+    throw InvalidConfigurationException();
+}
+
+void Setting::resourceTracker(){
+    cout << "SETTING GAME RESOURCE MARKERS!" << endl;
+    for(std::vector<Player>::iterator it = players->begin() ; it != players->end(); ++it){
+        it->resourceTracker()->score->insert(pair<ResourceTypes,std::uint_least16_t>(ResourceTypes::WHEAT,0));
+        it->resourceTracker()->score->insert(pair<ResourceTypes,std::uint_least16_t>(ResourceTypes::STONE,0));
+        it->resourceTracker()->score->insert(pair<ResourceTypes,std::uint_least16_t>(ResourceTypes::SHEEP,0));
+        it->resourceTracker()->score->insert(pair<ResourceTypes,std::uint_least16_t>(ResourceTypes::WOOD,0));
+    }
+    cout << "GAME RESOURCE MARKERS SET SUCCESSFULLY" << endl;
+}
```

Comment:
This will require adjustment when you make `resource_score` public.

Reply:
It will, which honestly I dont even understand why the teacher wants us to have a function to do that. It seems naturally to me that the ResourceTracker should clear itself and set a default state upon instantiation and computation

Reply:
I can make resource_score public. However, the Player instructions requires the ResourceTracker function and without the teacher adding more details as to its exact functionality i can do nothing but assume that it is a getter to the resource_score.